libwallet/assets/btc/dex-wallet.go

```diff
@@ -31,142 +35,291 @@ import (
 	"github.com/lightninglabs/neutrino/headerfs"
 )
 
-// DEXWallet wraps *wallet.Wallet and implements dexbtc.BTCWallet.
+// DEXWallet wraps *wallet.Wallet and implements dexbtc.Wallet.
 type DEXWallet struct {
-	w          *wallet.Wallet
-	acctNum    int32
-	spvService *btcChainService
+	w        *wallet.Wallet
+	acctNum  int32
+	cl       *btcChainService
+	syncData *SyncData
 }
 
-var _ dexbtc.BTCWallet = (*DEXWallet)(nil)
+var _ dexbtc.CustomWallet = (*DEXWallet)(nil)
 
 // NewDEXWallet returns a new *DEXWallet.
-func NewDEXWallet(w *wallet.Wallet, acctNum int32, nc *chain.NeutrinoClient) *DEXWallet {
+func NewDEXWallet(w *wallet.Wallet, acctNum int32, nc *chain.NeutrinoClient, syncData *SyncData) *DEXWallet {
 	return &DEXWallet{
 		w:       w,
 		acctNum: acctNum,
-		spvService: &btcChainService{
+		cl: &btcChainService{
```

Comment:
`btcChainService` isn't needed anymore. However, let's keep it because it has a `Peers()` method that is modified in #417 to be better than `nc.CS.Peers()`. But remove all other methods of `btcChainService`.

Reply:
Ok.